internal/api/external.go

```diff
@@ -95,10 +106,18 @@ func (a *API) ExternalProviderRedirect(w http.ResponseWriter, r *http.Request) e
 		InviteToken: inviteToken,
 		Referrer:    redirectURL,
 		FlowStateID: flowStateID,
-	})
+	}
+
+	if strings.Contains(r.URL.Path, "identities") {
```

Comment:
Would you mind matching on the actual URL here, like using the chi RoutePattern()

```suggestion
	if "/identities/authorize" == chi.RouteContext(r.Context()).RoutePattern() {
```